sklearn/datasets/base.py

```diff
@@ -382,11 +382,22 @@ def load_iris(return_X_y=False):
     if return_X_y:
         return data, target
 
+    feature_names = ['sepal length (cm)', 'sepal width (cm)',
+                     'petal length (cm)', 'petal width (cm)']
+
+    if as_frame:
+        try:
+            from pandas import Series, DataFrame
+            data_frame = DataFrame(data, columns=feature_names)
+            target_series = Series(target, name="class")
+            return data_frame, target_series
+        except ImportError:
+            pass
```

Comment:
If as_frame is true, then we should indeed raise the Importer of, not pass it

Reply:
i guess i am confused because i am thinking at the tests passing at the same time.